sycl/test-e2e/README.md

```diff
@@ -385,56 +385,56 @@ project must keep using `<sycl/sycl.hpp>` provided by the SYCL2020
 specification.**
 
 ### Compiling and executing tests on separate systems
-
 The execution of e2e tests can be separated into compilation and execution
 stages via the `test-mode` lit parameter. This allows us to reduce testing time
 by compiling tests on more powerful systems and reusing the binaries on other
 machines. By default the `test-mode` parameter is set to `full`, indicating that
 both stages will run. This parameter can be set to `build-only`, or `run-only`,
 to only run the compilation stage, or the execution stage respectively.
 
-**NOTE:** This feature is a work-in-progress and current limitations are
-expected to be addressed in the near future.
-
-The modes work as follow:
-
-* `--param test-mode=full`
-
-  This is the default mode tests run in. Tests are marked as unsupported if no
-  device on the machine can fulfill the `REQUIRES`/`UNSUPPORTED` statements. In
-  this mode all `RUN:` lines are executed normally, and two extra features are
-  added: the `build-and-run-mode` and `run-mode`.
+* `--param test-mode=run-only`
 
-  To make a test only run in `full` mode add a `REQUIRES: build-and-run-mode`
-  line.
+  In this mode, tests will not be compiled, they will only run. To do this only
+  the `RUN:` lines that contain a "run" expansion will be executed (`%{run}`,
+  `%{run-unfiltered-devices}`, or `%{run-aux}`). Since tests are not compiled in
+  this mode, for any test to pass the test binaries should already be in the
+  `test_exec_root` directory, either by having ran `full` or `build-only` modes
+  previously on the system, or having transferred the test binaries into that
+  directory. To mark a test as expected to fail only at run-time we can either
+  use the `run-mode` feature, which is only available in `run-only` and `full`
+  modes, or we can mark the test as `XFAIL` for a device-specific feature.
```

Comment:
That should be something along the lines of "XFAIL expression depending on a runtime (device-specific) feature".